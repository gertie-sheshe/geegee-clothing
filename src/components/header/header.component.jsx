import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { createStructuredSelector } from 'reselect';

import {auth} from '../../firebase/firebase.utils';
import { ReactComponent as Logo } from '../../assets/crown.svg';

import CartIcon from '../cart-icon/cart-icon.component';
import CartDropDown from '../cart-dropdown/cart-dropdown.component';

// Selectors
import {selectCartHidden} from '../../redux/cart/cart.selectors';
import {selectCurrentUser} from '../../redux/user/user.selectors';

import './header.styles.scss';

const Header = ({ currentUser, hidden }) => (
  <div className="header">
    <Link className="logo-container" to="/">
      <Logo className="logo" />
    </Link>
    <div className="options">
      <Link className="option" to="/shop">
        SHOP
      </Link>
      <Link className="option" to="/contact">
        CONTACT
      </Link>
      {currentUser ? (
        <div className="option" onClick={() => auth.signOut()} to="/contact">
          SIGN OUT
        </div>
      ) : (
        <Link to="/signin" className="option">
          SIGN IN
        </Link>
      )}
      <Link className="option" to=""></Link>
      <CartIcon />
    </div>
    {hidden ? null : <CartDropDown />}
  </div>
);

const mapStateToProps = createStructuredSelector({
    currentUser: selectCurrentUser,
    hidden: selectCartHidden
});

export default connect(mapStateToProps)(Header);
