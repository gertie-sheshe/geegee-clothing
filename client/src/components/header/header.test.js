import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './header.component';
import CartDropDown from '../cart-dropdown/cart-dropdown.component';

describe('Header Component Tests', () => {
  let wrapper;
  let mockSignOutStart;

  beforeEach(() => {
    mockSignOutStart = jest.fn();

    const mockProps = {
      hidden: true,
      currentUser: {
        uid: '123',
      },
      signOutStart: mockSignOutStart,
    };

    wrapper = shallow(<Header {...mockProps} />);
  });

  it('Should render Header component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('If currentUser is present', () => {
    it('Should render sign out link', () => {
      expect(
        wrapper
          .find('OptionLink')
          .at(2)
          .text(),
      ).toBe('SIGN OUT');
    });

    it('should call Sign out start method when link is created', () => {
      wrapper
        .find('OptionLink')
        .at(2)
        .simulate('click');
      expect(mockSignOutStart).toHaveBeenCalled();
    });
  });

  describe('if currentUser is null', () => {
    it('Should render sign in link', () => {
      const mockProps = {
        hidden: true,
        currentUser: null,
        signOutStart: mockSignOutStart,
      };

      const newWrapper = shallow(<Header {...mockProps} />);
      expect(
        newWrapper
          .find('OptionLink')
          .at(2)
          .text(),
      ).toBe('SIGN IN');
    });
  });

  describe('if hidden is true', () => {
    it('Should not render CartDropDown', () => {
      expect(wrapper.exists(CartDropDown)).toBe(false);
    });
  });

  describe('if hidden is false', () => {
    it('Should render CartDropdown', () => {
      const mockProps = {
        hidden: false,
        currentUser: null,
        signOutStart: mockSignOutStart,
      };

      const newWrapper = shallow(<Header {...mockProps} />);
      expect(newWrapper.exists(CartDropDown)).toBe(true);
    });
  });
});
