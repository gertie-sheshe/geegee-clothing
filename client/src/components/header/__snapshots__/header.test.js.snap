// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Tests Should render Header component 1`] = `ShallowWrapper {}`;
